jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamTablesResultSet.java

```diff
@@ -127,18 +127,12 @@ protected boolean doNextPage() throws SQLException {
   /**
    * Retrieve the databases in the Timestream instance to retrieve tables for.
    *
-   * @param database the specified database to use, may be null.
+   * @param schemaPattern the schema pattern to use to match database names, maybe null.
    * @return the databases in the Timestream instance to list tables for.
    * @throws SQLException if there is an error reading from Timestream.
    */
-  private Iterator<String> getDatabases(String database) throws SQLException {
-    if (null != database) {
-      LOGGER.debug("Specific database: \"{}\" provided for table retrieval.", database);
-      // A database was specified, so use that.
-      return ImmutableList.of(database).iterator();
-    }
-
-    try (ResultSet rs = new TimestreamDatabasesResultSet(connection)) {
+  private Iterator<String> getDatabases(String schemaPattern) throws SQLException {
+    try (ResultSet rs = new TimestreamSchemasResultSet(connection, schemaPattern)) {
```

Comment:
The schemaPattern (formerly `database`) is used in `TimestreamSchemasResultSet` to ensure that only tables from database that match `schemaPattern` are returned. Therefore the null check at the top was removed